### PR TITLE
JavaScript CapStone Project

### PR DESCRIPTION
**Pull request by faisalakb**
In this project, we accomplished the following tasks :100: 

 - Implemented Gitflow for version control. :+1: 
 - Set up Linters and Webpack for code quality and build management. :+1: 
 - Utilized the TV Maze API as the foundation for developing the web app. :+1: 
 - Integrated the Involvement API to record user interactions, such as likes and comments. :+1: 

    Created interfaces for the home page and comment popup. :+1: 

  **Home Page:** :1st_place_medal: 
       - On page load, the web app fetches data from the selected API to display a list of items. :+1: 
       - It also fetches data from the Involvement API to show item likes. :+1: 
       - When users click the "Hear" button for an item, the interaction is recorded in the Involvement API, and the screen is 
             updated accordingly. :+1: 
        - Clicking the "Comments" button triggers the appearance of the Comments popup. :+1: 
        - The home page header and navigation are designed to match the provided mockup. :+1: 
        - Similarly, the home page footer adheres to the mockup design. :+1: 
    **Comments Popup:** :1st_place_medal: 
        - When the popup loads, the web app fetches data from the selected API to display details about the selected item. :+1: 
        - It also fetches data from the Involvement API to show item comments. :+1: 
        - When users click the "Submit" button, the data is recorded in the Involvement API, and the screen is updated   
          accordingly. :+1: 
    **Counters:** :1st_place_medal: 
        -  All interfaces feature counters to display the number of items on the home page. :+1: 
        - The comment popup also shows the number of comments for the selected item. :+1:

### CODE REVIEW

**Review by AshakaE — COMMENTED** [review]
### Hi Team 🙋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request ❌ 

You have submitted a project, that does not follow the [group capstone guidelines for this project](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/code-review/articles/code_review_flow_group_projects.md)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: module/test/cardscounter.test.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    { title: 'Card 2' },
+    { title: 'Card 3' },
+  ];
+
+  // Act
+  cardscounter(cards);
+
+  // Assert
+  expect(document.querySelector('#cardNum').innerHTML).toBe('Shows(3)');
+});
```
Comment (abdulhamiid) [inline module/test/cardscounter.test.js:17]: - [ ] Please let' ensure that the `counter` function covers all the edge cases you can think about. (at least about the scenario in which there are 0 items on the page (counter function should return 0)🤓

- [ ] Anecdotally, it is important to test edges cases because this is where errors tend to arise 👍

File: README.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank the Microverse full-stack curriculum for the inspiration and guidance.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is under [MIT](https://github.com/faisalakb/Javascript-capston/blob/development/LICENSE) licensed.
```
Comment (abdulhamiid) [inline README.md:289]: Please fix the broken LICENSE link, let's avoid broken links in our readme for a professional repository 🛠 👍 

image

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
## Hi @faisalakb, @GhostEsso 👋, 



Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 2 above)
Comment (faisalakb) [reply README.md]: Resolve all the comments mention above

**Review by danmainah — APPROVED** [review]
Hi @faisalakb and @GhostEsso , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented the change requested to ensure the cardsCounter tests cover edge cases.
✔️ Implemented change requested to fix broken link ensuring the README is professionally written with all sections required.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GhostEsso** [comment]
@danmainah @AshakaE @abdulhamiid 
I would like to thank you, thank you for your advice and indications.
